galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.ceda.manager.ivt/src/main/java/dev/galasa/cicsts/ceda/manager/ivt/CedaManagerIVT.java

```diff
@@ -203,49 +157,90 @@ public void testResourceLibrary() throws TextNotFoundException, ErrorTextFoundEx
-      }
-      assertThat(result).isEqualTo(true);
-
-      // Checking if group delete works by discarding elements from CEMT and deleting group from CEDA, checking by installing group
-      if (result) {
-         result=false;
-        cics.ceda().deleteGroup(cedaTerminal, groupName);
-        cics.cemt().discardResource(cemtTerminal, resourceType, resourceName);
-        cics.cemt().discardResource(cemtTerminal, resourceType, resourceName2);
-        cics.cemt().discardResource(cemtTerminal, resourceType, resourceName3);
-        assertThatThrownBy(() -> {
-           cics.ceda().installGroup(cedaTerminal, groupName);
-        }).isInstanceOf(CedaException.class).hasMessageContaining("Problem determining the result from the CEDA command");
-
-      } else Fail.fail("CEDA Group Install/Delete failed");
-   }
-
-}
+           } else Fail.fail("Failed to intsall / delete resource");
+
+     } catch (CedaException | CemtException e) {
+        e.printStackTrace();
+     }
+  }
+
+  @Test
+  public void testResourceLibrary() throws TextNotFoundException, ErrorTextFoundException, Zos3270Exception, InterruptedException, CicstsManagerException {
+     String resourceType = "LIBRARY";
+     String resourceName = "lib1";
+     String groupName = "Test";
+     String resourceParameters = "DSNAME01(CTS.USER.APPL1.CICS.LOAD)";
+     boolean response = false;
+     try {
+        // Testing create and install resource by creating it, installing it and then checking if it appeared on CEMT
+        assertThat(cics.ceda().resourceExists(cedaTerminal, resourceType, resourceName, groupName)).isEqualTo(false);
+
+        cics.ceda().createResource(cedaTerminal, resourceType, resourceName, groupName, resourceParameters);
+        assertThat(cics.ceda().resourceExists(cedaTerminal, resourceType, resourceName, groupName)).isEqualTo(true);
+
+        cics.ceda().installResource(cedaTerminal, resourceType, resourceName, groupName);
+        if (cics.cemt().inquireResource(cemtTerminal, resourceType, resourceName) != null) {
+           response = true;
+        }
+        assertThat(response).isEqualTo(true);
+			
+        // If resource was installed successfully, then tests the delete method by discarding resource from CEMT, 
+        // deleting and then trying to install and checking if the resource appeared on CEMT
+
+        if (response) {
+           response=false;
+           cics.cemt().discardResource(cemtTerminal, resourceType, resourceName);
+
+           if (cics.cemt().inquireResource(cemtTerminal, resourceType, resourceName) == null) {
+              cics.ceda().deleteResource(cedaTerminal, resourceType, resourceName, groupName);
+              cics.ceda().installResource(cedaTerminal, resourceType, resourceName, groupName);
+              if (cics.cemt().inquireResource(cemtTerminal, resourceType, resourceName) == null) {
+                 response = true;
+              }
+              assertThat(response).isEqualTo(true);

```

Comment:
If we have just installed the resource shouldn't we have a response?

Reply:
Hi Will please can you clarify what you mean here?